tank_lab_to_nwb/convert_towers_task/towersnwbconverter.py

```diff
@@ -16,16 +17,9 @@ class TowersNWBConverter(NWBConverter):
         VirmenData=VirmenDataInterface,
     )
 
-    def __init__(self, **input_args):
-        self._recording_type = 'SpikeGLXRecording'
-        super().__init__(**input_args)
-
-    def get_recording_type(self):
-        return self._recording_type
-
     def get_metadata(self):
-        session_path = self.data_interface_objects['VirmenData'].input_args['folder_path']
-        subject_path, session_id = os.path.split(session_path)
+        file_path = Path(self.data_interface_objects['VirmenData'].input_args['file_path'])
+        session_id = file_path.stem
 
         session_name = os.path.splitext(session_id)[0]
         date_text = [name for name in session_name.split('_') if name.isdigit()][0]
```

Comment:
This way there is no need for `session_name`, we can get the date part directly from `session_id`
```suggestion
        date_text = [id_part for id_part in session_id.split('_') if id_part.isdigit()][0]
```